src/debug/dwarf/unit.go

```diff
@@ -62,12 +62,24 @@ func (d *Data) parseUnits() ([]unit, error) {
 		var n Offset
 		n, u.is64 = b.unitLength()
 		vers := b.uint16()
+		// Count the bytes of header read, not counting size
```

Comment:
Rather than counting bytes, I'd suggest following what parseTypes does. Here, just say "hdroff := b.off", then below say "b.bytes(int(n - (b.off - hdroff)))".